### PR TITLE
Support for adding custom files to the NPM install directory

### PR DESCRIPTION
**Pull request by etspaceman**

### CODE REVIEW

File: core/src/main/scala/io/chrisdavenport/npmpackage/sbtplugin/NpmPackagePlugin.scala [hunk 1]
```diff
@@ -372,11 +377,31 @@ object NpmPackagePlugin extends AutoPlugin {
...
+      npmPackageExtraFiles.value.foreach { f =>
+        val targetPath = (targetDir / f.name).toPath
+
+        Files.copy(
+          f.toPath,
+          targetPath,
+          StandardCopyOption.REPLACE_EXISTING
+        )
+        log.info(s"Wrote $f to $targetPath")
+      }
```
[the remark this answers is not in the document]
Comment (etspaceman) [reply core/src/main/scala/io/chrisdavenport/npmpackage/sbtplugin/NpmPackagePlugin.scala:396]: 🤔 Yeah I suppose we could support a directory. Let me play with that.

[the remark this answers is not in the document]
Comment (etspaceman) [reply core/src/main/scala/io/chrisdavenport/npmpackage/sbtplugin/NpmPackagePlugin.scala]: Updated

File: core/src/main/scala/io/chrisdavenport/npmpackage/sbtplugin/NpmPackagePlugin.scala [hunk 2]
```diff
@@ -172,11 +172,15 @@ object NpmPackagePlugin extends AutoPlugin {
     val npmPackageType: SettingKey[String] = 
       settingKey("The type of the package - defaults to 'commonjs' for ModuleKind.CommonJSModule or ModuleKind.NoModule, and 'module' for ModuleKind.ESModule")
 
+    lazy val npmPackageExtraFiles: SettingKey[Seq[File]] =
+      settingKey[Seq[File]]("Extra files to copy to the NPM install directory")
```
Comment (armanbilge) [inline core/src/main/scala/io/chrisdavenport/npmpackage/sbtplugin/NpmPackagePlugin.scala:176]: It's copying to `npmPackageOutputDirectory` right, so should we say the package output directory?

```suggestion
      settingKey[Seq[File]]("Extra files to copy to the NPM install directory")
```

**Review by armanbilge — COMMENTED** [review]
thanks! looks good, just a docs nitpick.

File: core/src/main/scala/io/chrisdavenport/npmpackage/sbtplugin/NpmPackagePlugin.scala (hunk 2 above)
Comment (etspaceman) [reply core/src/main/scala/io/chrisdavenport/npmpackage/sbtplugin/NpmPackagePlugin.scala]: Done

File: core/src/main/scala/io/chrisdavenport/npmpackage/sbtplugin/NpmPackagePlugin.scala [hunk 3]
```diff
@@ -172,11 +172,15 @@ object NpmPackagePlugin extends AutoPlugin {
...
 
+    lazy val npmPackageExtraFiles: SettingKey[Seq[File]] =
+      settingKey[Seq[File]]("Extra files to copy to the Npm package output directory")
+
     val npmPackage = taskKey[Unit]("Creates all files and direcories for the npm package")
 
     val npmPackageOutputJS = taskKey[File]("Write JS to output directory")
     val npmPackagePackageJson = taskKey[File]("Write Npm Package File to Directory")
     val npmPackageWriteREADME = taskKey[File]("Write README to the npm package")
+    val npmPackageWriteExtraFiles = taskKey[Unit]("Copy extra files to the NPM install directory")
```
Comment (armanbilge) [inline core/src/main/scala/io/chrisdavenport/npmpackage/sbtplugin/NpmPackagePlugin.scala:183]:
```suggestion
    val npmPackageWriteExtraFiles = taskKey[Unit]("Copy extra files to the NPM package output directory")
```

**Review by armanbilge — APPROVED** [review]

**Pull request merged**